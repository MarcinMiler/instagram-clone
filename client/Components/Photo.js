import React from 'react'
import { Image } from 'react-native'
import { Container } from '../Styled'

import Navbar from '../Components/Navbar'
import BottomBar from '../Components/BottomBar'
import Post from '../Components/Post'

const Photo = ({ photo, isLiked, likePhoto, navigation }) => (
    <Container>
        <Navbar back leftIcon="arrow-left" title="Photo" />
        <Post
            photo={photo}
            isLiked={isLiked}
            likePhoto={likePhoto}
            navigation={navigation}
        />
        <BottomBar />
    </Container>
)

export default Photo
