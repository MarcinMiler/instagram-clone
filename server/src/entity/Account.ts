import { Entity, Column, BaseEntity, PrimaryGeneratedColumn } from 'typeorm'

@Entity()
export class Account extends BaseEntity {
    @PrimaryGeneratedColumn() id: number

    @Column({ unique: true })
    email: string

    @Column() password: string
}
