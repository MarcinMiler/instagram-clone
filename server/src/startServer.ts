import 'reflect-metadata'
import { importSchema } from 'graphql-import'
import { GraphQLServer } from 'graphql-yoga'
import * as path from 'path'
import * as fs from 'fs'
import * as jwt from 'jsonwebtoken'
import { mergeSchemas, makeExecutableSchema } from 'graphql-tools'
import { GraphQLSchema } from 'graphql'
import { createTypeormConn } from './utils/createTypeormConn'
import SECRET from './utils/SECRET'

const Expo = require('expo-server-sdk')

const options = {
    port: process.env.NODE_ENV === 'test' ? 0 : 4000,
    endpoint: '/graphql',
    cors: {
        origin: '*'
    }
}

const addUser = (req: any, _: any, next: any) => {
    const token = req.headers.token
    if (token) {
        const user = jwt.verify(token, SECRET)
        req.user = user
    }
    next()
}

export const startServer = async () => {
    const schemas: GraphQLSchema[] = []
    const folders = fs.readdirSync(path.join(__dirname, './modules'))
    folders.forEach(folder => {
        const { resolvers } = require(`./modules/${folder}/resolvers`)
        const typeDefs = importSchema(
            path.join(__dirname, `./modules/${folder}/schema.graphql`)
        )
        schemas.push(makeExecutableSchema({ resolvers, typeDefs }))
    })

    const expo = new Expo()

    const server = new GraphQLServer({
        schema: mergeSchemas({ schemas }),
        context: (req: any) => {
            if (req.request.user) {
                return {
                    user: req.request.user.id,
                    expo
                }
            }
            return
        }
    })
    server.use(addUser)

    await createTypeormConn()

    const app = await server.start(options)

    return app
}
