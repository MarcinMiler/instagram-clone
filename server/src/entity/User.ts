import {
    Entity,
    PrimaryColumn,
    Column,
    OneToMany,
    BaseEntity,
    CreateDateColumn,
    ManyToMany,
    JoinTable,
    RelationCount
} from 'typeorm'
import { Photo } from './Photo'

@Entity()
export class User extends BaseEntity {
    @PrimaryColumn() id: number

    @CreateDateColumn() date: Date

    @Column() email: string

    @Column() fullname: string

    @Column() username: string

    @Column({ nullable: true })
    bio: string

    @OneToMany(() => Photo, photo => photo.user)
    photos: Photo[]

    @ManyToMany(() => User, user => user.following)
    @JoinTable()
    followers: User[]

    @ManyToMany(() => User, user => user.followers)
    following: User[]

    @RelationCount((user: User) => user.photos)
    photosCount: number

    @RelationCount((user: User) => user.followers)
    followersCount: number

    @RelationCount((user: User) => user.following)
    followingCount: number
}
