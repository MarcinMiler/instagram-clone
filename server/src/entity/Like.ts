import {
    Entity,
    PrimaryGeneratedColumn,
    ManyToOne,
    BaseEntity,
    Column,
    OneToOne,
    JoinColumn
} from 'typeorm'
import { Photo } from './Photo'
import { User } from './User'

@Entity()
export class Like extends BaseEntity {
    @PrimaryGeneratedColumn() id: number

    @Column({ nullable: true })
    photoId: number

    @Column({ nullable: true })
    userId: number

    @ManyToOne(() => Photo, photo => photo.likes)
    photo: Photo

    @OneToOne(() => User)
    @JoinColumn()
    user: User
}
