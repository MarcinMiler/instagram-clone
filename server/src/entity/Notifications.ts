import {
    Entity,
    Column,
    BaseEntity,
    PrimaryGeneratedColumn,
    OneToOne,
    JoinColumn
} from 'typeorm'
import { User } from './User'

@Entity()
export class Notification extends BaseEntity {
    @PrimaryGeneratedColumn() id: number

    @Column({ nullable: true })
    userId: number

    @Column({ nullable: true })
    photoId: number

    @Column() sendTo: number

    @Column() message: string

    @OneToOne(() => User)
    @JoinColumn()
    user: User
}
