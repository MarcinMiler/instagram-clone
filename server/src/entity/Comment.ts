import {
    Entity,
    Column,
    PrimaryGeneratedColumn,
    BaseEntity,
    ManyToOne,
    OneToOne,
    JoinColumn,
    OneToMany,
    RelationCount
} from 'typeorm'
import { Photo } from './Photo'
import { User } from './User'
import { Like } from './Like'

@Entity()
export class Comment extends BaseEntity {
    @PrimaryGeneratedColumn() id: number

    @Column() text: string

    @Column({ nullable: true })
    photoId: number

    @Column({ nullable: true })
    userId: number

    @ManyToOne(() => Photo, photo => photo.comments)
    photo: Photo

    @OneToMany(() => Like, like => like.comment)
    likes: Like[]

    @RelationCount((comment: Comment) => comment.likes)
    likesCount: number

    @OneToOne(() => User)
    @JoinColumn()
    user: User
}
