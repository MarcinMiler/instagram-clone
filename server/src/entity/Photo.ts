import {
    Entity,
    PrimaryGeneratedColumn,
    Column,
    ManyToOne,
    BaseEntity,
    OneToMany
} from 'typeorm'
import { User } from './User'
import { Like } from './Like'

@Entity()
export class Photo extends BaseEntity {
    @PrimaryGeneratedColumn() id: number

    @Column() url: string

    @Column({ nullable: true })
    userId: number

    @ManyToOne(() => User, user => user.photos)
    user: User

    @OneToMany(() => Like, like => like.photo)
    likes: Like[]
}
