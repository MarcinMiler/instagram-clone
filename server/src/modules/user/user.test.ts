import { request, GraphQLClient } from 'graphql-request'
import { startServer } from '../../startServer'
import {
    loginMutationWithToken,
    registerMutation,
    searchUsersQuery,
    follow,
    unfollowMutation,
    userQuery
} from '../../utils/mutations'
import { User } from '../../entity/User'
import { Photo } from '../../entity/Photo'

let getHost = () => ''
let token = ''

const email = 'm@m.com'
const password = 'mm'
const fullname = 'Marcin Miler'
const username = 'Marcinek'

beforeAll(async () => {
    const app = await startServer()
    const { port } = app.address()
    getHost = () => `http://127.0.0.1:${port}/graphql`

    await request(
        getHost(),
        registerMutation(email, password, fullname, username)
    )

    await request(
        getHost(),
        registerMutation('a@a.com', 'aa', 'Angelika Miler', 'Angelaaa')
    )

    const login: any = await request(
        getHost(),
        loginMutationWithToken(email, password)
    )
    token = login.login.token
})

describe('Query user', async () => {
    it('should return user', async () => {
        const client = new GraphQLClient(getHost(), {
            headers: {
                token
            }
        })

        await Photo.create({
            url: 'lul.jpg',
            userId: 1
        }).save()

        const response = await client.request(userQuery(1))

        expect(response).toMatchSnapshot()
    })
})
