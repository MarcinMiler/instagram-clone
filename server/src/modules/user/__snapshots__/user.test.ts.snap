// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Query user should return user 1`] = `
Object {
  "user": Object {
    "email": "m@m.com",
    "fullname": "Marcin Miler",
    "id": "1",
    "photos": Array [
      Object {
        "id": "1",
        "url": "lul.jpg",
      },
    ],
    "username": "Marcinek",
  },
}
`;
